L63-dotnet-grpc-web.md

```diff
@@ -0,0 +1,82 @@
+gRPC-Web in-process proxy and client for grpc-dotnet
+----
+* Author(s): James Newton-King
+* Approver: Jan Tattermusch & Wenbo Zhu
+* Status: In review
+* Implemented in: https://github.com/grpc/grpc-dotnet/pull/695
+* Last updated: 2020-01-10
+* Discussion at: TBD
+
+## Abstract
+
+gRPC-Web for grpc-dotnet adds:
+
+* An in-process proxy for gRPC hosted on ASP.NET Core, allowing the server to support gRPC-Web calls.
+* A gRPC-Web outgoing proxy for the .NET gRPC client, allowing the client to make gRPC-Web calls.
+
+## Background
+
+gRPC requires HTTP/2 features that are not available in browser HTTP clients. [gRPC-Web](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md) modifies the gRPC specification to allow gRPC to be callable from the browser.
+
+Traditionally gRPC-Web support is added to a service via a proxy. Browsers call the proxy and the proxy translates the gRPC-Web call to gRPC HTTP/2 and forwards it on to the service implentation. This works, but there is effort, complexity and performance costs associated with adding a proxy to a solution. An in-process proxy allows for gRPC-Web to be supported by adding additional packages and startup configuration to an existing app. Being able to support gRPC-Web with an in-process proxy removes many of the costs of the traditional gRPC-Web proxy.
+
+Calling gRPC-Web is typically done using the JavaScript gRPC-Web library. This works for JavaScript solutions, but the modern .NET ecosystem allows for .NET applications to run in the browser using Web Assembly. A gRPC-Web .NET client is needed for this scenario.
+
+The grpc-dotnet gRPC server is built on ASP.NET Core and hosted on Kestrel, a full-featured, multi-purpose HTTP/1.1 and HTTP/2 server. ASP.NET Core has a [request pipeline that an application can plug middleware into](https://docs.microsoft.com/aspnet/core/fundamentals/middleware) to process the request and response. ASP.NET Core also supports configuring CORS for an application.
+
+The grpc-dotnet gRPC client is built on HttpClient, a full-featured, multi-purpose HTTP/1.1 and HTTP/2 client. HttpClient has an [outgoing request pipe that an application can plug handlers into](https://docs.microsoft.com/dotnet/api/system.net.http.httpmessagehandler).
```

Comment:
Can HTTP/1.1 and HTTP/2 being ran on the same connection? Or the new implementation needs to manage two types of `subchannel`?

Reply:
nope, you either have HTTP/1.1 connection or you have HTTP/2.
From users perspective, you either configure a Channel to create HTTP1.1 connections and use grpc-web  or you configure it to use regular grpc wire protocol over HTTP2.